BrainPortal/app/controllers/resource_usage_controller.rb

```diff
@@ -81,5 +79,54 @@ def base_scope() #:nodoc:
     scope
   end
 
+  # Create list of RUs visible to current user with restricted scope
+  def restrict_scope(maintype,scope,params) #:nodoc:
+
+    @base_scope   = base_scope
+                    .where('resource_usage.type' => @maintype)
+    if  @maintype == 'SpaceResourceUsageForUserfile'
+
+      if params[:deleted_items]
+        @base_scope = @base_scope.includes(:userfile).where(:userfiles => {id: nil})
+      end
+
+      if params[:negative_file_delta]
+        @base_scope = @base_scope.where('resource_usage.value < 0')
+      end
+
+      if params[:positive_file_delta]
+        @base_scope = @base_scope.where('resource_usage.value >= 0')
+      end
+
+    else
+      if params[:deleted_items]
+        @base_scope = @base_scope = @base_scope.includes(:cbrain_task).where(:cbrain_task => {id: nil})
+      end
+    end
+
+    if params[:data]
```

Comment:
Why did you call the structure `data` ?!? Isn't this one of the _noise words_ that you mentioned in your presentation? The default is even `date_range` !